rules/test.bzl

```diff
@@ -57,20 +57,25 @@ def _ios_test(name, test_rule, test_suite_rule, apple_library, infoplists_by_bui
         else:
             ios_test_kwargs["runner"] = runner
 
-    library = apple_library(name = name, namespace_is_module_name = False, platforms = {"ios": ios_test_kwargs.get("minimum_os_version")}, **kwargs)
-
-    fw_name = name + ".framework_middleman"
-    framework_middleman(name = fw_name, framework_deps = library.deps, tags = ["manual"])
-    ios_test_kwargs["frameworks"] = [fw_name]
-
     # Deduplicate against the test deps
     if ios_test_kwargs.get("test_host", None):
         host_args = [ios_test_kwargs["test_host"]]
     else:
         host_args = []
+    library = apple_library(name = name, namespace_is_module_name = False, platforms = {"ios": ios_test_kwargs.get("minimum_os_version")}, **kwargs)
+
+    # Setup framework middlemen - need to process deps and libs
+    fw_name = name + ".framework_middleman"
+    framework_middleman(name = fw_name, framework_deps = kwargs.get("deps", []) + library.lib_names, tags = ["manual"])
+    frameworks = [fw_name] + ios_test_kwargs.pop("frameworks", [])
+
+    dep_name = name + ".dep_middleman"
+    dep_middleman(name = dep_name, deps = kwargs.get("deps", []) + library.lib_names, tags = ["manual"], test_deps = host_args)
```

Comment:
Nit: this is very similar between test/framework/app, consider extracting to a single macro and consuming in all places

Reply:
➕  the main caveat is that we currently feed it `test_deps` to govern the bundling of a test bundle. I've been considering this as well: making an `apple_bundling_rule` factory where we can build out rules or macros for that. This might be something for a future effort where we add support for all the rules like extensions, widgets, etc.

Reply:
SGTM!